### PR TITLE
Make it easier to get the token authority from the response.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/TokenResponse.java [hunk 1]
```diff
@@ -111,6 +111,23 @@
     @SerializedName("id_token")
     private String mIdToken;
 
+    private transient String mTokenAuthority;
+
+    /**
+     * Get the authority that issued this token.
+     * @return the authority that issued this token.
+     */
+    public String getAuthority() {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/TokenResponse.java:120]: Are there usages as of now?

File: common/src/main/java/com/microsoft/identity/common/exception/ClientException.java [hunk 2]
```diff
@@ -27,6 +27,11 @@
     public static final String sName = ClientException.class.getName();
     private static final long serialVersionUID = -2318746536590284648L;
 
+    /**
+     * Indicates that a token request returned an unexpected authority.
+     */
+    public static final String UNEXPECTED_AUTHORITY = "unexpected_authority_token_request";
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/exception/ClientException.java:33]: Are there usages as of now?

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/exception/ClientException.java:33]: This shouldn't be here.

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/TokenResponse.java (hunk 1 above)
Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/TokenResponse.java:120]: No, but this is an attempt to replace some of the circumlocution that we're going through to figure out authorities.

**Pull request closed without merging**